velox/vector/FlatVector.h

```diff
@@ -394,8 +394,15 @@ class FlatVector final : public SimpleVector<T> {
     return true;
   }
 
+  // Acquire ownership for any string buffer that appears in source, the
+  // function does nothing if the vector type is not Varchar or Varbinary.
+  // The function throws if input encoding is lazy.
   void acquireSharedStringBuffers(const BaseVector* source);
 
+  // Acquire ownership for any string buffer that appears in source or any
+  // of its children recursively. The function throws if input encoding is lazy.
+  void acquireSharedStringBuffersRecursive(const BaseVector* source);
```

Comment:
do we need 2 separate implementations? in other words, is there any functionality that relies on throwing an error when calling acquireSharedStringBuffers() with anything other than flat or constant? If not, maybe we can just have one single API.

Reply:
I did that initially check V2, then masha suggested that I have two interfaces.